packages/ember-data/lib/system/store.js

```diff
@@ -1864,7 +1861,11 @@ Store = Service.extend({
     @return {String} if the adapter can generate one, an ID
   */
   _normalizeTypeKey: function(key) {
-    return camelize(singularize(key));
+    // Delegate to the container for normalization. The container
+    // requires a ':' to normalize so we add `model:` then slice off
+    // the first 6 characters to remove the `model:` in the return
+    // value
+    return this.container.normalize('model:' + key).slice(6);
```

Comment:
we should add `normalizeType('model')` to the container, which then delegates to the resolver.